src/components/AeInput.vue

```diff
@@ -8,49 +8,80 @@
       :slot="slot"
       :name="slot"
     />
-    <input
-      :id="id"
-      v-focus.lazy="autofocus"
+    <div
       slot-scope="{ setFocus, id }"
-      :value="value"
-      v-bind="$attrs"
-      @focus="setFocus(true)"
-      @blur="setFocus(false)"
-      @input="$emit('input', $event.target.value)"
+      class="labeled-input"
     >
+      <input
+        :id="id"
+        v-focus.lazy="autofocus"
+        :value="formatDisplayValue(value)"
+        v-bind="$attrs"
+        @focus="setFocus(true)"
+        @blur="setFocus(false)"
+        @input="$emit('input', formatEmitValue($event.target.value))"
+      >
+      <label v-if="nameInput">
+        {{ AENS_DOMAIN }}
+      </label>
+    </div>
   </AeInputWrapper>
 </template>
 
 <script>
 import { focus } from 'vue-focus';
 import AeInputWrapper from './AeInputWrapper.vue';
+import { AENS_DOMAIN } from '../lib/constants';
 
 export default {
   components: { AeInputWrapper },
   directives: { focus },
   props: {
     value: { type: String, default: '' },
     autofocus: Boolean,
+    nameInput: Boolean,
```

Comment:
I propose to create a new component as `AeInputAmount`, instead of mixing different use cases in a single component

also, you may extract common styles of `<input>` between all these components